scipy/optimize/_linprog.py

```diff
@@ -587,7 +588,14 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
     if x0 is not None and meth != "revised simplex":
         warning_message = "x0 is used only when method is 'revised simplex'. "
         warn(warning_message, OptimizeWarning)
-
+        
+    if meth in {'revised simplex', 'simplex'}:
+        if sps.issparse(A_eq):
+            A_eq = A_eq.todense()
+        if sps.issparse(A_ub):
+            A_ub = A_ub.todense()
```

Comment:
I didn't realize that `todense` returns a `np.matrix` rather than `np.ndarray` (gh-14494). If we go this route, let's make these `toarray`.

Reply:
Yes I thought that was a little weird, I haven't seen np.matrix used much

Reply:
Very timely. I'm preparing a PR that removes all unreasonable uses of `.todense` from the codebase.